src/mbgl/util/camera.cpp

```diff
@@ -0,0 +1,260 @@
+#include "camera.hpp"
+#include <cassert>
+#include <cmath>
+#include <mbgl/map/camera.hpp>
+#include <mbgl/math/log2.hpp>
+#include <mbgl/util/constants.hpp>
+#include <mbgl/util/geo.hpp>
+#include <mbgl/util/projection.hpp>
+
+namespace mbgl {
+
+using vec2 = std::array<double, 2>;
```

Comment:
shall be public type imo^

Reply:
there's a vec2 also in RenderLocationIndicatorLayer, would this being public clash with that?

Reply:
why both of them cannot use the same generic type?

Reply:
There shouldn't be any clashes with the vec2 in `RenderLocationIndicatorImpl`. I would actually like to see us implementing separate types for vector classes like done in the location indicator rendering (instead of using type aliases) in the future.